src/overlays/actors/ovl_En_Test/z_en_test.c

```diff
@@ -28,22 +28,247 @@ const ActorInit En_Test_InitVars = {
+        func_8018219C(&sp38, &thisx->shape.rot, 1);
+        thisx->world.rot = thisx->shape.rot;
+        this->unk_20A = func_800C9BB8(&globalCtx->colCtx, thisx->floorPoly, sp34);
+    }
+
+    func_80183430(&this->skeletonInfo, &gameplay_keep_Blob_06EB70, &gameplay_keep_Blob_06BB0C, this->unk_178,
+                  this->unk_1C0, NULL);
+    func_801834A8(&this->skeletonInfo, &gameplay_keep_Blob_06BB0C);
+    this->skeletonInfo.frameCtrl.unk_10 = 9.0f;
+    func_80862B70(this->unk_20C);
+}
+
+void EnTest_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnTest* this = THIS;
+
+    func_8018349C(&this->skeletonInfo);
+}
+
+void EnTest_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnTest* this = THIS;
+    s32 i;
+
+    this->unk_208 = this->skeletonInfo.frameCtrl.unk_10;
+
+    if (func_80183DE0(&this->skeletonInfo) && (this->actor.parent == NULL) && (this->actor.params != -1)) {
+        this->unk_209++;
+        if (this->unk_209 > 20) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+
+    if ((this->actor.params != -1) && (this->unk_208 >= 2)) {
+        if (!(this->unk_174 & 1)) {
+            this->unk_174 |= 1;
+
+            for (i = 0; i < ARRAY_COUNT(this->unk_20C); i++) {
+                func_80862CBC(this->unk_20C, &this->actor.world.pos);
+            }
+        }
+    }
+    func_80862EDC(this->unk_20C);
+}
+
+s32 func_808634B8(GlobalContext* globalCtx, SkeletonInfo* skeletonInfo, s32 limbIndex, Gfx** dList, u8* flags,
+                  Actor* thisx, Vec3f* scale, Vec3s* rot, Vec3f* pos) {
+    EnTest* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    if ((this->actor.params == -1) && ((limbIndex == 1) || (limbIndex == 2))) {
+        *dList = NULL;
+    }
+
+    if ((this->actor.params == -2) && (limbIndex == 3)) {
+        *dList = NULL;
+    }
+
+    if (limbIndex == 3) {
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0x80, 0, 0, 0, ((20 - this->unk_209) * 255.0f) / 20.0f);
+        gDPSetEnvColor(POLY_XLU_DISP++, 0, 0, 0, ((20 - this->unk_209) * 255.0f) / 20.0f);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
+
+    return true;
+}
+
+void EnTest_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnTest* this = THIS;
+    Mtx* sp28;
+    s32 sp2C = this->unk_208 - 1;
+
+    if (sp2C >= 29) {
+        sp2C = 29;
+    }
+
+    if ((this->unk_20A == 15) || (this->unk_20A == 14)) {
+        AnimatedMat_DrawStep(globalCtx, Lib_SegmentedToVirtual(gameplay_keep_Matanimheader_06B730), sp2C);
+    } else {
+        AnimatedMat_DrawStep(globalCtx, Lib_SegmentedToVirtual(gameplay_keep_Matanimheader_06B6A0), sp2C);
+    }
+
+    sp28 = GRAPH_ALLOC(globalCtx->state.gfxCtx, ALIGN16(this->skeletonInfo.unk_18->unk_1 * sizeof(Mtx)));
+
+    if (sp28 != NULL) {
+        func_8012C2DC(globalCtx->state.gfxCtx);
+        EnTest_OverrideKeyframeDraw(globalCtx, &this->skeletonInfo, sp28, func_808634B8, NULL, thisx);
```

Comment:
```suggestion
    if (sp28 != NULL) {
        func_8012C2DC(globalCtx->state.gfxCtx);
        func_8018450C(globalCtx, &this->skeletonInfo, sp28, EnTest_OverrideKeyframeDraw, NULL, thisx);
```
I think you misread what AngheloAlf meant. `func_808634B8` should be renamed.